src/basic/parse-util.c

```diff
@@ -268,6 +269,62 @@ int parse_range(const char *t, unsigned *lower, unsigned *upper) {
         return 0;
 }
 
+int parse_errno(const char *t) {
+        uint8_t u;
+        int r;
+
+        assert(t);
+
+        r = errno_from_name(t);
+        if (r > 0)
+                return r;
+
+        r = safe_atou8(t, &u);
```

Comment:
hmm, why parse this as u8? right now we have 133 errnos on Linux, but in theory there could be more, no? Or there an API that enshrines somehow that this must be 8bit?

If there isn't I'd probably permit any 15bit value >= 0...

Reply:
@poettering Thank you for your comment. I've limited the range to u8 based on the suggestion by @keszybz:
>  exit(3) says that the return value is ~~ORed~~ ANDed with 255.

However, errno is sometimes used as the `exist()` status, but not always. And `SCMP_ACT_ERRNO()` takes `uint16_t`. So, I think 0 to 65535 is the valid range here. What do you think?

Reply:
exit() and errno are different concepts.

process exit status values are 0..255, that's true. But the codes returned there are usually EXIT_SUCCESS and EXIT_FAILURE, as defined by libc, which are a completely different thing than errno, which is afaics always stored as "int", and uses the Exxx constants. Or to say this differently, both libc's EXIT_FAILURE and libc's EPERM map to "1", but are very different things...

Reply:
hmm, so, if libseccomp limits things to 0…65535 we should not offer more, that's true.

So I checked the kernel itself now. Internally, the kernel actually defines MAX_ERRNO to be 4095. See linux/err.h. I figure that's what we should limit ourselves to then, too.

I.e. use safe_atoi(), then check that what we parsed is neither negative nor >= 4096...

Reply:
OK. To use `MAX_ERRNO`, should I copy linux/err.h to src/shared/linux/err.h? or define `MAX_ERRNO` somewhere e.g. missing.h? or just hard-code the value in `parse_errno()`?

Reply:
it's an internal header of the kernel, we shouldn't make direct use of it in userspace. Hence, maybe just add it to errno-list.h, and maybe even name it differently, to avoid future clashes. 

Note that there's actually errno_max() defined already (though unused), which returns the max defined errno systemd knows about. Given that it is unused, maybe we should just remove it, and then define ERRNO_MAX in its place to the fixed 4095 value?

Reply:
OK. Thank you for your kind advice.